sdk/core/azure-core-amqp/src/amqp/message_sender.cpp

```diff
@@ -302,16 +305,28 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
           {
             if (deliveryStatus.IsNull())
             {
-              deliveryStatus = m_savedMessageError;
+              error = m_savedMessageError;
+            }
+            else
+            {
+              auto deliveryStatusAsList{deliveryStatus.AsList()};
+              Models::AmqpValue firstState{deliveryStatusAsList[0]};
```

Comment:
Is it guaranteed that the list will be non-empty?

Reply:
I sure hope so. The uAMQP code seems to indicate it. I'll add some defensive checks too.